src/main/java/graphql/schema/AbstractGraphQLType.java

```diff
@@ -0,0 +1,32 @@
+package graphql.schema;
+
+import static graphql.Assert.assertNotNull;
+
+public class AbstractGraphQLType implements GraphQLType {
+    private final String name;
+
+    public AbstractGraphQLType(String name) {
+        assertNotNull(name, "name can't null");
+        this.name = name;
+    }
+
+    public String getName() {
+        return name.equals("") ? null : name;
+    }
```

Comment:
I don't think that `GraphQLModifiedType` implementations are ever asked for their name.
Of course that's a little bit hard to see from the interface hierarchy, because the base interface declares `getName()`.

Technically we could return anything, it's never used anywhere. Of course that does not hinder any client to ask for name. The reference implementation uses `GraphQLNamedType` for this, I guess if we did that it would be clearer. Not sure if it's really worth it, though.